homeassistant/components/camera/__init__.py

```diff
@@ -788,18 +788,27 @@ async def async_refresh_providers(self, *, write_state: bool = True) -> None:
         providers or inputs to the state attributes change.
         """
         old_provider = self._webrtc_provider
-        new_provider = await self._async_get_supported_webrtc_provider(
-            async_get_supported_provider
-        )
-
         old_legacy_provider = self._legacy_webrtc_provider
+        new_provider = None
         new_legacy_provider = None
-        if new_provider is None:
-            # Only add the legacy provider if the new provider is not available
-            new_legacy_provider = await self._async_get_supported_webrtc_provider(
-                async_get_supported_legacy_provider
+
+        # Skip all providers if the camera has a native WebRTC implementation
+        if not (
+            self._webrtc_sync_offer
+            or type(self).async_handle_async_webrtc_offer
+            != Camera.async_handle_async_webrtc_offer
```

Comment:
We can determine this state up front in the unit method and cache it as an instance attribute, similar to `_webrtc_sync_offer`.